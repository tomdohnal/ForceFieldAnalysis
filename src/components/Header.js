// @flow
import React, { Component } from 'react';
import { Divider } from 'semantic-ui-react';

import { Box, Text } from './common';
import { COLORS, HEADER_HEIGHT } from '../constants/styles';

type State = {
  showAbout: boolean,
}

class Header extends Component<any, State> {
  state = {
    showAbout: false,
  };

  onAboutLinkClick = () => {
    this.setState({ showAbout: true });
  };

  onCloseAboutLinkClick = () => {
    this.setState({ showAbout: false });
  };

  render() {
    return (
      <Box height={HEADER_HEIGHT} paddingTop="16px" textAlign="center" position="relative">
        <Text fontSize="24px">Force Field Analysis Tool</Text>
        {this.state.showAbout ?
          <Box
            position="absolute"
            zIndex="999"
            backgroundColor={COLORS.WHITE}
            width="500px"
            right="16px"
            padding="16px"
            textAlign="initial"
            border={`2px ${COLORS.BLUE} solid`
            }
          >
            <Text
              onClick={this.onCloseAboutLinkClick}
              position="absolute"
              right="16px"
              top="8px"
              clickable
            >Close
            </Text>
            <Text fontSize="32px" marginTop="16px">What is Force Field Analysis?</Text>
            <Divider hidden />
            <Text fontSize="18px">Force Field Analysis is a decision-making tool. Originally created in 1940s by psychologist Kurt Lewin, it is widely used in business for making decisions. The idea is to create driving (also called helping) forces that are driving change and hindering (also called resisting) forces that are hindering change. You can attribute different strengths to each force. Once you've done with that, you can compare the sum of your driving forces and your hindering forces and decide if you should go or not go for change!</Text>
          </Box>
          :
          <Text
            onClick={this.onAboutLinkClick}
            position="absolute"
            right="20px"
            clickable
          >About
          </Text>
        }
      </Box>
    );
  }
}

export default Header;
