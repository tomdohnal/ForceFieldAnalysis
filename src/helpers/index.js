// @flow
type Action = {
  type: string,
  payload?: any,
}

export type Reducer<S, A: Action> = (S, A) => S;

export type ReduxHandlers<S, A: Action> = {
  [key: string]: Reducer<S, A>,
}

export const createReducer =
  <S, A: Action>(initialState: S, handlers: ReduxHandlers<S, A>) =>
    (state: S = initialState, action: A) => (
      Object.prototype.hasOwnProperty.call(handlers, action.type) ?
        handlers[action.type](state, action) : state
    );

export const moveCaretAtTheEnd = (e: SyntheticEvent<HTMLInputElement>) => {
  if (e.target instanceof HTMLInputElement) {
    const { value } = e.target;
    e.target.value = '';
    e.target.value = value;
  }
};
