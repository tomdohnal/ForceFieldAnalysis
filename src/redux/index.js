// @flow
import { combineReducers, createStore } from 'redux';
import _ from 'lodash';

import { loadState, saveState } from './localStorage';

import forcesReducer, { type Forces } from '../ducks/forces';
import changeReducer, { type Change } from '../ducks/change';
import modeReducer, { type Mode } from '../ducks/mode';

export type ReduxState = {
  forces: Forces,
  change: Change,
  mode: Mode,
}

export const RESET_STATE = 'RESET_STATE';

export type ResetStateAction = {
  type: typeof RESET_STATE
}

export const resetState = (): ResetStateAction => ({ type: RESET_STATE });

const appReducer = combineReducers({
  forces: forcesReducer,
  change: changeReducer,
  mode: modeReducer,
});

const rootReducer = (state, action) => {
  if (action.type === RESET_STATE) {
    return appReducer({}, action);
  }

  return appReducer(state, action);
};

const persistedState = loadState();

export const store = createStore(rootReducer, persistedState);

store.subscribe(_.throttle(() => {
  saveState(store.getState());
}, 1000));
