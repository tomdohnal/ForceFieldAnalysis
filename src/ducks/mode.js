// @flow
import { createReducer } from '../helpers';

export const SWITCH_TO_EDIT_MODE: string = 'MODES/SWITCH_TO_EDIT_MODE';
export const SWITCH_TO_COMPLETE_MODE: string = 'MODES/SWITCH_TO_COMPLETE_MODE';

export const EDIT_MODE = 'EDIT_MODE';
export const COMPLETE_MODE = 'COMPLETE_MODE';

export type Mode = typeof EDIT_MODE | typeof COMPLETE_MODE;

export type SwitchToEditModeAction = {
  type: typeof SWITCH_TO_EDIT_MODE,
  payload: {
    mode: typeof EDIT_MODE,
  }
}

export type SwitchToCompleteModeAction = {
  type: typeof SWITCH_TO_COMPLETE_MODE,
  payload: {
    mode: typeof COMPLETE_MODE,
  }
}

export const switchToEditMode = (): SwitchToEditModeAction => ({
  type: SWITCH_TO_EDIT_MODE,
  payload: {
    mode: EDIT_MODE,
  },
});

export const switchToCompleteMode = (): SwitchToCompleteModeAction => ({
  type: SWITCH_TO_COMPLETE_MODE,
  payload: {
    mode: COMPLETE_MODE,
  },
});

export default createReducer(EDIT_MODE, {
  [SWITCH_TO_EDIT_MODE]: (state: Mode, action: SwitchToEditModeAction) => (
    action.payload.mode
  ),
  [SWITCH_TO_COMPLETE_MODE]: (state: Mode, action: SwitchToCompleteModeAction) => (
    action.payload.mode
  ),
});
