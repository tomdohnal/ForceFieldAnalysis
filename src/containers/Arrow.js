// @flow
import React, { Component } from 'react';
import { Form, Input, Button, Icon, Divider } from 'semantic-ui-react';
import EditIcon from 'react-icons/lib/ti/edit';
import { connect } from 'react-redux';

import { MAX_STRENGTH, MIN_STRENGTH } from '../constants/index';
import { COLORS } from '../constants/styles';
import { type Force, deleteForce, setForceName, increaseForce, decreaseForce } from '../ducks/forces';
import { Text, Box } from '../components/common/index';
import { Rectangle, Triangle } from '../components/svg/index';
import TextBubble from '../components/TextBubble';

type Props = {
  color: string,
  force: Force,
  deleteForce: (id: string) => void,
  setForceName: (id: string, name: string) => void,
  increaseForce: (id: string) => void,
  decreaseForce: (id: string) => void,
}

type State = {
  editArrow: boolean,
}

class Arrow extends Component<Props, State> {
  state = {
    editArrow: false,
  };

  onDeleteArrowIconClick = () => {
    this.props.deleteForce(this.props.force.id);
  };

  onIncreaseArrowIconClick = () => {
    if (this.props.force.strength !== MAX_STRENGTH) {
      this.props.increaseForce(this.props.force.id);
    }
  };

  onDecreaseArrowIconClick = () => {
    if (this.props.force.strength !== MIN_STRENGTH) {
      this.props.decreaseForce(this.props.force.id);
    }
  };

  onEditArrowInputChange = (event, data) => {
    this.props.setForceName(this.props.force.id, data.value);
  };

  onEditArrowNameClick = () => {
    this.setState({ editArrow: true });
  };

  onEditArrowNameFormSubmit = () => {
    this.setState({ editArrow: false });
  };

  render() {
    const { force: { driving, strength, name }, color } = this.props;

    const arrowUnitWidth = 50;
    const arrowWidth = strength * arrowUnitWidth;
    const arrowHeight = 80;
    const rectangleHeight = arrowHeight - 40;
    const triangleWidth = 36;

    return (
      <Box>
        <TextBubble color={color} left={!driving}>
          {this.state.editArrow ?
            <Form onSubmit={this.onEditArrowNameFormSubmit}>
              <Input autoFocus size="small" value={name} onChange={this.onEditArrowInputChange} action>
                <input style={{ width: 'inherit' }} />
                <Button size="small">Save changes</Button>
              </Input>
            </Form>
            :
            <Box>
              <Text fontSize="15px">{name}</Text>
              <EditIcon
                size="18"
                style={{ verticalAlign: 'top', marginLeft: '4px', cursor: 'pointer' }}
                onClick={this.onEditArrowNameClick}
              />
            </Box>
          }
        </TextBubble>
        <Box position="relative">
          <Icon
            name="delete"
            size="big"
            style={{
              position: 'absolute',
              top: 'calc(50% - 14px)',
              color: COLORS.RED,
              cursor: 'pointer',
              [driving ? 'right' : 'left']: `${arrowWidth}px`,
            }}
            onClick={this.onDeleteArrowIconClick}
          />
          <svg height={arrowHeight} width={arrowWidth} viewBox={`0 0 ${arrowWidth} ${arrowHeight}`}>
            <Rectangle
              width={arrowWidth - triangleWidth}
              height={rectangleHeight}
              color={color}
              x={!driving ? triangleWidth : 0}
              y={(arrowHeight - rectangleHeight) / 2}
            />
            <Triangle
              width={triangleWidth}
              height={arrowHeight}
              x={`${!driving ? triangleWidth : arrowWidth - triangleWidth},0`}
              y={`${!driving ? triangleWidth : arrowWidth - triangleWidth},${arrowHeight}`}
              z={`${!driving ? 0 : arrowWidth},${arrowHeight / 2}`}
              color={color}
            />
          </svg>
          <Box
            position="absolute"
            top={`${arrowHeight - ((arrowHeight - rectangleHeight) / 2)}px`}
            right={driving && `${triangleWidth}px`}
            left={!driving && `${triangleWidth}px`}
          >
            <Icon
              name="plus"
              size="large"
              disabled={strength === MAX_STRENGTH}
              style={{ cursor: `${strength < MAX_STRENGTH ? 'pointer' : 'initial'}` }}
              onClick={this.onIncreaseArrowIconClick}
            />
            <Text fontSize="18px">{strength}</Text>
            <Icon
              name="minus"
              size="large"
              disabled={strength === MIN_STRENGTH}
              style={{ cursor: `${strength > MIN_STRENGTH ? 'pointer' : 'initial'}` }}
              onClick={this.onDecreaseArrowIconClick}
            />
          </Box>
        </Box>
        <Divider hidden />
      </Box>
    );
  }
}

const actionCreators = { deleteForce, setForceName, increaseForce, decreaseForce };

export default connect(undefined, actionCreators)(Arrow);
