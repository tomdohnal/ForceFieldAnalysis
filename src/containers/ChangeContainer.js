// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Input, Button } from 'semantic-ui-react';

import { COLORS } from '../constants/styles';
import type { ReduxState } from '../redux/index';
import { setChangeName, setChangeDescription, type Change } from '../ducks/change';
import { Box, Text } from '../components/common/index';

type Props = {
  change: Change,
  onNameChange: (name: string) => void,
  onDescriptionChange: (description: string) => void,
};

type State = {
  editChangeName: boolean,
  editChangeDescription: boolean,
};

class ChangeContainer extends Component<Props, State> {
  state = {
    editChangeName: !this.props.change.name,
    editChangeDescription: !this.props.change.description,
  };

   onButtonClick = editChangeBoolean => (): void => {
     this.setState({ [editChangeBoolean]: false });
   };

   renderForm(onChange: (value: string) => void, onButtonClick: (value: string) => void) {
     return (
       <Box>
         <Input onChange={onChange} />
         <Button onClick={onButtonClick} />
       </Box>
     );
   }

   render() {
     const { change: { name, description }, onNameChange, onDescriptionChange } = this.props;

     return (
       <Box padding="20px 16px" color={COLORS.BLUE} height="100%">
         {this.state.editChangeName ?
           this.renderForm(onNameChange, this.onButtonClick('editChangeName')) :
           <Text>{name}</Text>
         }
         {this.state.editChangeDescription ?
           this.renderForm(onDescriptionChange, this.onButtonClick('editChangeDescription')) :
           <Text>{description}</Text>
         }
       </Box>
     );
   }
}

const mapStateToProps = (state: ReduxState) => ({
  change: state.change,
});

const mapDispatchToProps = dispatch => ({
  onNameChange: (event, data) => (dispatch(setChangeName(data.value))),
  onDescriptionChange: (event, data) => (dispatch(setChangeDescription(data.value))),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeContainer);
