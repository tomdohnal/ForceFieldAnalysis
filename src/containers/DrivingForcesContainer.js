// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Input, Button } from 'semantic-ui-react';

import DrivingForceArrow from '../components/DrivingForceArrow';
import { addDrivingForce, type Forces } from '../ducks/forces';

type State = {
  newDrivingForceName: string,
  newDrivingForceError: boolean,
  newDrivingForceErrorText: string,
};

type Props = {
  forces: Forces,
  addDrivingForce: (name: string, strength: number) => void,
};

class DrivingForcesContainer extends Component<Props, State> {
  state = {
    newDrivingForceName: '',
    newDrivingForceError: false,
    newDrivingForceErrorText: '',
  };

  onNewDrivingForceInputChange = (e) => {
    this.setState({
      newDrivingForceName: e.target.value,
      newDrivingForceError: false,
      newDrivingForceErrorText: '',
    });
  };

  onNewDrivingForceButtonClick = () => {
    if (!this.state.newDrivingForceName) {
      this.setState({
        newDrivingForceError: true,
        newDrivingForceErrorText: 'Enter the description for the force.',
      });
    } else {
      this.setState({
        newDrivingForceName: '',
      });

      this.props.addDrivingForce(this.state.newDrivingForceName, 2);
    }
  };

  renderForces() {
    return this.props.forces.map(force => (
      <DrivingForceArrow strength={force.strength} key={force.id} />));
  }

  render() {
    return (
      <div>
        {this.renderForces()}
        <Input
          onChange={this.onNewDrivingForceInputChange}
          value={this.state.newDrivingForceName}
          error={this.state.newDrivingForceError}
        />
        <Button onClick={this.onNewDrivingForceButtonClick}>New driving force</Button>
        <span>{this.state.newDrivingForceErrorText}</span>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  forces: state.forces.filter(force => force.driving === true),
});

export default connect(mapStateToProps, { addDrivingForce })(DrivingForcesContainer);
