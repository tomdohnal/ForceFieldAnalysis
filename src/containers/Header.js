// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Grid } from 'semantic-ui-react';

import { Box, Text } from '../components/common/index';
import { COLORS, HEADER_HEIGHT } from '../constants/styles';
import AboutBox from '../components/AboutBox';
import { resetState, type ResetStateAction, type ReduxState } from '../redux';
import { switchToCompleteMode, switchToEditMode, type SwitchToCompleteModeAction, type SwitchToEditModeAction, type Mode, EDIT_MODE } from '../ducks/mode';

type Props = {
  resetState: () => ResetStateAction,
  switchToCompleteMode: () => SwitchToCompleteModeAction,
  switchToEditMode: () => SwitchToEditModeAction,
  appMode: Mode,
}

type State = {
  showAbout: boolean,
}

export class Header extends Component<Props, State> {
  state = {
    showAbout: false,
  };

  onResetButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.resetState();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onMakeDecisionButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.switchToCompleteMode();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onEditForcesButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.switchToEditMode();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onAboutLinkClick = () => {
    this.setState({ showAbout: true });
  };

  onCloseAboutLinkClick = () => {
    this.setState({ showAbout: false });
  };

  render() {
    return (
      <Box height={HEADER_HEIGHT} paddingTop="16px">
        <Grid columns={3}>
          <Grid.Row>
            <Grid.Column>
              <Box paddingLeft="16px">
                <Button
                  onClick={this.onResetButtonClick}
                  style={{ color: COLORS.WHITE, background: COLORS.RED }}
                >
                  Reset analysis
                </Button>
                {this.props.appMode === EDIT_MODE ?
                  <Button onClick={this.onMakeDecisionButtonClick}>Make decision</Button>
                  :
                  <Button onClick={this.onEditForcesButtonClick}>Edit forces</Button>
                }
              </Box>
            </Grid.Column>
            <Grid.Column>
              <Box textAlign="center">
                <Text fontSize="24px">Force Field Analysis Tool</Text>
              </Box>
            </Grid.Column>
            <Grid.Column floated="right">
              {this.state.showAbout ?
                <AboutBox onCloseLinkClick={this.onCloseAboutLinkClick} />
                :
                <Text
                  onClick={this.onAboutLinkClick}
                  position="absolute"
                  right="30px"
                  clickable
                >About
                </Text>
              }
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Box>
    );
  }
}

export const mapStateToProps = (state: ReduxState) => ({ appMode: state.mode });

export default connect(
  mapStateToProps, { resetState, switchToCompleteMode, switchToEditMode },
)(Header);
