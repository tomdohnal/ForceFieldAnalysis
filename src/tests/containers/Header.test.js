import React from 'react';
import { shallow } from 'enzyme';

import { Header, mapStateToProps } from '../../containers/Header';
import { EDIT_MODE, COMPLETE_MODE, switchToEditMode as switchToEditModeOriginal } from '../../ducks/mode';
import { mockShallowComponentMethod } from '../helpers';
import { store } from '../../redux';

describe('Header', () => {
  const resetState = jest.fn();
  const switchToCompleteMode = jest.fn();
  const switchToEditMode = jest.fn();

  const ShallowHeader = shallow(
    <Header
      appMode={EDIT_MODE}
      resetState={resetState}
      switchToCompleteMode={switchToCompleteMode}
      switchToEditMode={switchToEditMode}
    />,
  );

  describe('rendering', () => {
    it('renders correctly in edit mode', () => {
      ShallowHeader.setProps({ appMode: EDIT_MODE });

      expect(ShallowHeader).toMatchSnapshot();
    });

    it('renders correctly in complete mode', () => {
      ShallowHeader.setProps({ appMode: COMPLETE_MODE });

      expect(ShallowHeader).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('onResetButtonClick calls reset state prop', () => {
      ShallowHeader.instance().onResetButtonClick({});

      expect(resetState).toBeCalled();
    });

    it('onMakeDecisionButtonClick calls switchToCompleteMode', () => {
      ShallowHeader.instance().onMakeDecisionButtonClick({});

      expect(switchToCompleteMode).toBeCalled();
    });

    it('onEditForcesButtonClick calls switchToEditMode', () => {
      ShallowHeader.instance().onEditForcesButtonClick({});

      expect(switchToEditMode).toBeCalled();
    });

    it('onAboutLinkClick sets showAbout state to true', () => {
      ShallowHeader.setState({ showAbout: false });

      ShallowHeader.instance().onAboutLinkClick();

      expect(ShallowHeader.state().showAbout).toBeTruthy();
    });

    it('onCloseAboutLinkClick sets showAbout state to false', () => {
      ShallowHeader.setState({ showAbout: true });

      ShallowHeader.instance().onCloseAboutLinkClick();

      expect(ShallowHeader.state().showAbout).toBeFalsy();
    });
  });

  describe('method triggering', () => {
    it('calls onResetButtonClick when reset button is clicked', () => {
      mockShallowComponentMethod(ShallowHeader, 'onResetButtonClick');

      ShallowHeader.find('Button').at(0).simulate('click');

      expect(ShallowHeader.instance().onResetButtonClick).toBeCalled();
    });

    it('calls onAboutLinkClick when about is clicked', () => {
      mockShallowComponentMethod(ShallowHeader, 'onAboutLinkClick');

      ShallowHeader.find('text').at(1).simulate('click');

      expect(ShallowHeader.instance().onAboutLinkClick).toBeCalled();
    });

    it('calls onMakeDecisionButtonClick when the make decision button is clicked', () => {
      ShallowHeader.setProps({ appMode: EDIT_MODE });

      mockShallowComponentMethod(ShallowHeader, 'onMakeDecisionButtonClick');

      ShallowHeader.find('Button').at(1).simulate('click');

      expect(ShallowHeader.instance().onMakeDecisionButtonClick).toBeCalled();
    });

    it('calls onEditForcesButtonClick when the make decision button is clicked', () => {
      ShallowHeader.setProps({ appMode: COMPLETE_MODE });

      mockShallowComponentMethod(ShallowHeader, 'onEditForcesButtonClick');

      ShallowHeader.find('Button').at(1).simulate('click');

      expect(ShallowHeader.instance().onEditForcesButtonClick).toBeCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('picks mode from the global state and maps it to appMode', () => {
      store.dispatch(switchToEditModeOriginal());

      const props = mapStateToProps(store.getState());

      expect(props).toEqual({ appMode: EDIT_MODE });
    });
  });
});
