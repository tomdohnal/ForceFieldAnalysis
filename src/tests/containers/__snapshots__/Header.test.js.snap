// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendering renders correctly in complete mode 1`] = `
<boxStyles
  height="60px"
  paddingTop="16px"
>
  <Grid
    columns={3}
  >
    <GridRow>
      <GridColumn>
        <boxStyles
          paddingLeft="16px"
        >
          <Button
            as="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#dc2f2f",
                "color": "#fff",
              }
            }
          >
            Reset analysis
          </Button>
          <Button
            as="button"
            onClick={[Function]}
          >
            Edit forces
          </Button>
        </boxStyles>
      </GridColumn>
      <GridColumn>
        <boxStyles
          textAlign="center"
        >
          <text
            fontSize="24px"
          >
            Force Field Analysis Tool
          </text>
        </boxStyles>
      </GridColumn>
      <GridColumn
        floated="right"
      >
        <text
          clickable={true}
          onClick={[Function]}
          position="absolute"
          right="30px"
        >
          About
        </text>
      </GridColumn>
    </GridRow>
  </Grid>
</boxStyles>
`;

exports[`Header rendering renders correctly in edit mode 1`] = `
<boxStyles
  height="60px"
  paddingTop="16px"
>
  <Grid
    columns={3}
  >
    <GridRow>
      <GridColumn>
        <boxStyles
          paddingLeft="16px"
        >
          <Button
            as="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#dc2f2f",
                "color": "#fff",
              }
            }
          >
            Reset analysis
          </Button>
          <Button
            as="button"
            onClick={[Function]}
          >
            Make decision
          </Button>
        </boxStyles>
      </GridColumn>
      <GridColumn>
        <boxStyles
          textAlign="center"
        >
          <text
            fontSize="24px"
          >
            Force Field Analysis Tool
          </text>
        </boxStyles>
      </GridColumn>
      <GridColumn
        floated="right"
      >
        <text
          clickable={true}
          onClick={[Function]}
          position="absolute"
          right="30px"
        >
          About
        </text>
      </GridColumn>
    </GridRow>
  </Grid>
</boxStyles>
`;
